### PR TITLE
Add URL endpoint to grab a snapshot of the config.git repository (#2188)

### PR DESCRIPTION
**Pull request by ketan**
Git push has been explicitly disabled.

Usage -

```bash
$ git clone https://ci.example.com/go/api/config-repository[.git]
```

### CODE REVIEW

**Comment by zabil** [comment]
How's authentication handled?

**Comment by ketan** [comment]
> How's authentication handled?

https://github.com/gocd/gocd/pull/2240/files#diff-10f894ef12c1ff6e9c8dac675693e03cR268

**Comment by ketan** [comment]
> How's authentication handled?
> https://github.com/gocd/gocd/pull/2240/files#diff-10f894ef12c1ff6e9c8dac675693e03cR268

On second thought, there's a typo on that line, will wait for other comments.

File: server/webapp/WEB-INF/applicationContext-acegi-security.xml [hunk 1]
```diff
@@ -263,6 +266,8 @@
                 /api/admin/material_test=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
                 /api/admin/internal/pipelines=ROLE_SUPERVISOR,ROLE_GROUP_SUPERVISOR
                 /api/admin/**=ROLE_SUPERVISOR
+                /api/config-repository/**=ROLE_SUPERVISOR
+                /api/config-repository.git/**=ROLE_SUPERVISOR
```
Comment (zabil) [inline server/webapp/WEB-INF/applicationContext-acegi-security.xml]: If we name the end point ```/admin/config.git``` we won't need this entry.

Comment (ketan) [inline server/webapp/WEB-INF/applicationContext-acegi-security.xml]: It conflicts with [this rule](https://github.com/gocd/gocd/blob/453f3d60368484f094ca0e6633d060a6042ba1f9/server/webapp/WEB-INF/urlrewrite.xml#L93)

**Pull request merged**